misc: v25.0.0.0 release notes

erin-le opened the pull request.

erin-le committed:
misc: draft of v25.0.0.0 release notes

erin-le committed:
misc: revisons to v25.0 release notes

giactra commented on RELEASE-NOTES.md:
```diff
@@ -1,3 +1,209 @@
+
+### Arm
+
+- Added support for architectural extensions: `FEAT_FP16`
```
Hi, @erin-le , to be consistent with previous release, I would re-arrange the Arm section accordingly:

## ArmISA changes/improvements
### Architectural extensions
Architectural support for the following extensions:
* FEAT_FP16 ([#2071](https://github.com/gem5/gem5/pull/2071))
* FEAT_FHM ([#2287](https://github.com/gem5/gem5/pull/2287))
* FEAT_FRINTTS ([#2287](https://github.com/gem5/gem5/pull/2287))
* FEAT_S1PIE ([#1858](https://github.com/gem5/gem5/pull/1858))

### Bugfixes
* The following syscalls have been added in SE mode
  * clone3 (syscall 435) ([#1913](https://github.com/gem5/gem5/pull/1913))

### ArmPMU improvements
#### Cache Events
Add support for Cache PMU events. By hooking the PMU to l1d/l1i and l2 caches it is now possible to monitor their activities through performance counters (https://github.com/gem5/gem5/pull/1439)

#### Stat support
The ArmPMU has been enhanced to dump PMU counters as if they were common statistics. In this way a user can observe PMU events without requiring explicit programming from the guest application
(see https://github.com/gem5/gem5/pull/2271 for further details)

giactra commented on RELEASE-NOTES.md:
```diff
@@ -1,3 +1,209 @@
+extensions, bug fixes, and hypervisor extension support for RISC-V.
+- **Python Utilities**: Introduction of `gem5term`, an m5term replacement in
+Python, and the `hypercall_external_signal` utility for interacting with running
+simulations via hypercalls.
```
I believe the addition of OptionalParam and DictParam should be here as they are major additions to the user APIs

giactra reviewed (changes requested):
Hi @erin-le, please see my comments below

erin-le committed:
misc: v25.0 release notes - change Arm, add Optional/DictParam

powerjg commented on RELEASE-NOTES.md:
```diff
@@ -1,3 +1,225 @@
+
+## RISC-V
+
+- Added support for **hypervisor extension (H)** ([#1387](https://github.com/gem5/gem5/pull/1387)).
```
```suggestion
- Added support for **hypervisor extension (H)** ([#1387](https://github.com/gem5/gem5/pull/1387)).
  - **CAVEAT**: This support only works for atomic mode. Timing mode implementation is in development.
```

BobbyRBruce committed:
misc: Update RELEASE-NOTES.md

Co-authored-by: Jason Lowe-Power <jason@lowepower.com>

mattsinc commented on RELEASE-NOTES.md:
```diff
@@ -1,3 +1,226 @@
+([#1976](https://github.com/gem5/gem5/pull/1976)), improved dispatch scheduling
+([#2163](https://github.com/gem5/gem5/pull/2163)).
+- Added RLC queues to checkpoint ([#1999](https://github.com/gem5/gem5/pull/1999)),
+added pagetable walker buffer ([#2162](https://github.com/gem5/gem5/pull/2162)),
```
@tomxia: did any of your other commits get integrated in, in time for this release?  I don't think so, but want to double-check

TomXia replied:
I didn't see #1954 in the release notes, which fixed an assertion failure in VIPERSequencer.cc. But I think it has been integrated.

abmerop committed:
misc: Fill in GPU changes a bit more

dhschall commented on RELEASE-NOTES.md:
```diff
@@ -1,3 +1,226 @@
+- Enhanced load-store queue latency modeling ([#1926](https://github.com/gem5/gem5/pull/1926)).
+
+### Branch Prediction
+
```
```suggestion

- Support of speculative update of TAGE-SC-L
([#1854](https://github.com/gem5/gem5/pull/1854)) 
```

erin-le committed:
misc: update branch predictor section of v25.0 release notes

Co-authored-by: David Schall <david-schall@outlook.de>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

mattsinc committed:
Update RELEASE-NOTES.md

Fix typo in v24.0.0.0 release notes.

erin-le committed:
misc: additional updates to RELEASE-NOTES.md

This commit adds the change to the naming of Multisim's redirected
output files to the release notes. It also adds a note about the
addition of a SIGCONT signal handler to gem5.
